assert/utils.go

```diff
@@ -959,6 +959,61 @@ func formatNumericComparisonError(actual, expected interface{}, operation string
 	return msg.String()
 }
 
+// formatBeWithinError returns an error message when `actual` is outside `expected ± tolerance`.
+func formatBeWithinError[T Float](actual, expected, tolerance T) string {
+	diff := T(math.Abs(float64(actual - expected)))
+
+	actualF := float64(actual)
+	expectedF := float64(expected)
+	diffF := float64(diff)
+	toleranceF := float64(tolerance)
+
+	format := chooseFormat(actualF, expectedF, diffF, toleranceF)
+
+	var msg strings.Builder
+
+	msg.WriteString(fmt.Sprintf(
+		"Expected value to be within ±"+format+" of "+format+", but it was not:\n",
+		toleranceF, expectedF))
+	msg.WriteString(fmt.Sprintf("  Actual:    "+format+"\n", actualF))
+	msg.WriteString(fmt.Sprintf("  Expected:  "+format+"\n", expectedF))
+	msg.WriteString(fmt.Sprintf("  Tolerance: ±"+format+"\n", toleranceF))
+	msg.WriteString(fmt.Sprintf("  Difference: "+format, diffF))
+
+	if toleranceF > 0 {
+		excess := ((diffF - toleranceF) / toleranceF) * 100
+		if excess > 0 {
+			msg.WriteString(fmt.Sprintf(" (%.2f%% greater than tolerance)", excess))
+		}
```

Comment:
What about something like this


```suggestion
		excess := ((diffF - toleranceF) / toleranceF) 
		switch {
		case excess > 2:
			msg.WriteString(fmt.Sprintf(" (%.2f× greater than tolerance)", excess))
		case excess > 0:
			msg.WriteString(fmt.Sprintf(" (%.2f%% greater than tolerance)", 100 * excess))
		}
```

Because when you reach +100% non math people are lost.

And when you reach 10x, percentage looks abnormal